internal/dag/gatewayapi_processor.go

```diff
@@ -112,6 +112,14 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 						}
 					}
 				}
+			} else {
+				// Config is nil so validate the default which is mode: terminate
+				// Check for TLS on the Gateway.
+				if listenerSecret = p.validGatewayTLS(listener); listenerSecret == nil {
+					// If TLS was configured on the Listener, but it's invalid, don't allow any
+					// routes to be bound to this listener since it can't serve TLS traffic.
+					continue
+				}
```

Comment:
Related to @sunjayBhatia's comment, I'd do this a little differently for clarity, something like:

```go
case gatewayapi_v1alpha1.TLSProtocolType:
    if listener.TLS == nil {
        // log error and continue since .TLS is required if protocol is "TLS"
    }

    // if we get here we know listener.TLS is non-nil
    // proceed to checking tlsConfig.Mode
    if listener.TLS.Mode != nil {
        // ...
    }
```

So basically put the nil-check at the top, and no need to call into `p.validGatewayTLS(...)` if we know `listener.TLS` is nil

Reply:
Should be similar to how the [`HTTPSProtocolType` block](https://github.com/projectcontour/contour/blob/436f5dcedf980d60fe16ba0bbda6ba2701059690/internal/dag/gatewayapi_processor.go#L84-L95) is structured

Reply:
👍